recipe/meta.yaml

```diff
@@ -201,14 +201,14 @@ outputs:
     # uses lapack {{ version }}, not {{ blas_major }}
     script: install_blas_devel.sh   # [unix]
     build:
-      string: "{{ build_num }}_{{ build_string_platform }}_{{ blas_impl }}"
+      string: "{{ build }}_{{ build_string_platform }}_{{ blas_impl }}"
     requirements:
       host:
         - openblas   0.3.27  # [blas_impl == "openblas"]
-        - mkl-devel  2024.1  # [blas_impl == "mkl"]
+        - mkl-devel          # [blas_impl == "mkl"]
       run:
         - openblas   0.3.27  # [blas_impl == "openblas"]
-        - mkl-devel  2024.1  # [blas_impl == "mkl"]
+        - mkl-devel          # [blas_impl == "mkl"]
```

Comment:
AFAIU we cannot remove the pin for run here, because the run-export of `mkl-devel` is major-only and thus not equivalent.